### ISSUE TITLE
build helm with cgo_enabled on macOS

### ISSUE THREAD

**Issue by cpboyd** [Homebrew/homebrew-core#110500]
### `brew gist-logs <formula>` link OR `brew config` AND `brew doctor` output

```shell
➜  ~ brew config
HOMEBREW_VERSION: 3.6.0-24-g5524b30
ORIGIN: https://github.com/Homebrew/brew
HEAD: 5524b302d83a54660c49e850b3bc1558371c2fbc
Last commit: 5 days ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 05757fb2f3ad1512ac0fc12dcc838de983aa3dc1
Core tap last commit: 4 days ago
Core tap branch: master
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 10
HOMEBREW_NO_ANALYTICS: set
Homebrew Ruby: 2.6.8 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: 10-core 64-bit arm_firestorm_icestorm
Clang: 13.1.6 build 1316
Git: 2.37.3 => /opt/homebrew/bin/git
Curl: 7.79.1 => /usr/bin/curl
macOS: 12.6-arm64
CLT: 13.4.0.0.1.1651278267
Xcode: 13.4.1
Rosetta 2: false
➜  ~ brew doctor
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Some installed formulae are deprecated or disabled.
You should find replacements for the following formulae:
  gofish
  kubernetes-cli@1.22
```

### Verification

- [X] I ran `brew update` and am still able to reproduce my issue.
- [X] I have resolved all warnings from `brew doctor` and that did not fix my problem.

### What were you trying to do (and why)?

Helm should resolve URLs properly on macOS. (Prior to a change in 3.8.2 that disabled CGO, it did.)

Per https://github.com/helm/helm/issues/10874#issuecomment-1244761342, the interim solution until Go has a fix for https://github.com/golang/go/issues/12524 is to build with `CGO_ENABLED=1`

### What happened (include all command output)?

Helm doesn't resolve URLs with custom nameservers (e.g. private VPNs)

### What did you expect to happen?

Helm should properly resolve all URLs.

### Step-by-step reproduction instructions (by running `brew` commands)

```shell
brew install helm
```

**Comment by carlocab** [comment]
Could you please open a pull request for this?

You should already have a copy of this repository at `$(brew --repository homebrew/core)`.

Thanks.

**Comment by thesuperzapper** [comment]
@cpboyd I am slightly confused by this issue, are you proposing that we update our binaries for homebrew to be built with `CGO_ENABLED=1`?

If so, then I fully support this (and I think we should make this a P0), because helm effectively does not work properly on MacOS right now when installed from `brew install helm`.

**Comment by joejulian** [comment]
Opened a PR to allow CGO_ENABLED to be overridden [here](https://github.com/Homebrew/homebrew-core/blob/6a6df2d38de0761e09094724da509d44dc404253/Formula/helm.rb#L25)

**Comment by cpboyd** [comment]
Closed as [Go#12524](https://github.com/golang/go/issues/12524) should be resolved in v1.20
